packages/lib-user/src/components/UserHome/components/RecentProjects/RecentProjectsContainer.js

```diff
@@ -1,90 +1,56 @@
-  const authorization = `Bearer ${token}`
-  try {
-    const query = {
-      sort: '-updated_at',
-      user_id: user.id
-    }
-    const response = await panoptes.get('/project_preferences', query, { authorization })
-    if (response.ok) {
-      const projectPreferencesUserHasClassified =
-        response.body.project_preferences
-          .filter(preference => preference.activity_count > 0)
-      return projectPreferencesUserHasClassified
-    }
-    return []
-  } catch (error) {
-    console.error(error)
-    throw error
-  }
-}
-
 function RecentProjectsContainer({ authUser }) {
-  // Get user's project preference.activity_count for 10 most recently classified projects
-  const cacheKey = {
-    name: 'user-project-preferences',
-    userId: authUser.id
+  const recentProjectsQuery = {
+    project_contributions: true,
+    order_project_contributions_by: 'recents',
+    period: 'day'
   }
+
   const {
-    data: projectPreferences,
-    isLoading: preferencesLoading,
-    error: preferencesError
-  } = useSWR(cacheKey, fetchUserProjectPreferences, SWROptions)
+    data: stats,
+    isLoading: statsLoading,
+    error: statsError
+  } = useStats({ sourceId: authUser?.id, query: recentProjectsQuery })
 
-  // Get more info about each project and attach it to correct projectPreference object
-  const recentProjectIds = projectPreferences?.map(
-    preference => preference.links.project
-  )
+  // limit to 20 projects fetched from panoptes
+  const contributions = stats?.project_contributions.slice(0, 20)
+  const projectIds = contributions?.map(project => project.project_id)
+
+  // Get more info about each project
   const {
     data: projects,
     isLoading: projectsLoading,
     error: projectsError
   } = usePanoptesProjects({
     cards: true,
-    id: recentProjectIds?.join(',')
+    id: projectIds?.join(',')
   })
-  
-  // Attach project object to each project preference
-  let projectPreferencesWithProjectObj
-  if (projects?.length) {
-    projectPreferencesWithProjectObj = projectPreferences
-      .map(preference => {
-        const matchedProjectObj = projects.find(
-          project => project.id === preference.links?.project
-        )
 
-        if (matchedProjectObj) {
-          preference.project = matchedProjectObj
+  // Attach project info to each contribution stat (see similar behavior in TopProjects)
+  let recentProjects = []
+
+  if (projects?.length && contributions?.length) {
+    recentProjects = contributions
+      .map(projectContribution => {
+        const projectData = projects?.find(
+          project => project.id === projectContribution.project_id.toString()
+        )
+        return {
+          count: projectContribution.count,
+          ...projectData
         }
-        return preference
       })
-      .filter(preference => preference?.project?.slug)
-      .slice(0, 10)
+      .filter(project => project?.id) // exclude private or deleted projects
```

Comment:
I don't think it's an issue, but noting if I have 20 projects per line 19, but then private/deleted projects are filtered out here, then there might be less than 20 or an inconsistent number of projects for some users.

Reply:
Agreed. I think this is fine for the homepage, but we'll need to discuss handling private and/or deleted projects when creating an All Projects page with 20-per-page pagination.